frontends/concrete-python/tests/execution/test_matmul.py

```diff
@@ -140,19 +149,43 @@ def lhs_function(x):
     @fhe.compiler({"x": "encrypted"})
     def rhs_function(x):
         return np.matmul(lhs_cst, x)
-
-    lhs_inputset = [np.random.randint(minimum, maximum, size=lhs_shape) for i in range(100)]
-    rhs_inputset = [np.random.randint(minimum, maximum, size=rhs_shape) for i in range(100)]
+    
+    lhs_inputset = [rgen(lhs_shape) for i in range(100)]
+    rhs_inputset = [rgen(rhs_shape) for i in range(100)]
 
     lhs_operator_circuit = lhs_operator.compile(lhs_inputset, configuration)
     rhs_operator_circuit = rhs_operator.compile(rhs_inputset, configuration)
     lhs_function_circuit = lhs_function.compile(lhs_inputset, configuration)
     rhs_function_circuit = rhs_function.compile(rhs_inputset, configuration)
 
-    lhs_sample = np.random.randint(minimum, maximum, size=lhs_shape)
-    rhs_sample = np.random.randint(minimum, maximum, size=rhs_shape)
+    lhs_sample = rgen(lhs_shape)
+    rhs_sample = rgen(rhs_shape)
 
     helpers.check_execution(lhs_operator_circuit, lhs_operator, lhs_sample)
     helpers.check_execution(rhs_operator_circuit, rhs_operator, rhs_sample)
     helpers.check_execution(lhs_function_circuit, lhs_function, lhs_sample)
     helpers.check_execution(rhs_function_circuit, rhs_function, rhs_sample)
+
+    # Matmul of clear values and encrypted matrices
+    @fhe.compiler({"x": "encrypted", "y": "clear"})
+    def lhs_operator_clear(x, y):
+        return x @ y
+
+    # Matmul of two encrypted matrices
+    @fhe.compiler({"x": "encrypted", "y": "encrypted"})
+    def enc_function_xy(x, y):
+        return np.matmul(x, y)
+
+    # Put all the dual operand functions in a list
+
+    # lhs_operator_clear, rhs_operator_clear, 
+    dual_operand_functions = [lhs_operator_clear, enc_function_xy]
+
+    # Compile each dual operand function and test it on random data
+    for func in dual_operand_functions:
+        dual_op_inputset = [(rgen(lhs_shape), rgen(rhs_shape)) for i in range(100)]
+        dual_op_circuit = func.compile(dual_op_inputset, configuration)        
+
+        lhs_sample, rhs_sample = rgen(lhs_shape), rgen(rhs_shape)
+        
+        helpers.check_execution(dual_op_circuit, func, [lhs_sample, rhs_sample])
```

Comment:
I feel like these belong to another test. See https://github.com/zama-ai/concrete/blob/main/frontends/concrete-python/tests/execution/test_add.py

Reply:
Maybe a refactor even